docs/installation/server-config/index.rst

```diff
@@ -10,25 +10,21 @@ Configuring KSQL Server
     ccloud-ksql
     avro-schema
 
-KSQL configuration parameters can be set for Admin Client, KSQL Query, KSQL Server, Kafka Streams, and Kafka Clients. The
-KSQL Server parameters are configured via the ``/etc/ksql/ksql-server.properties`` file. 
+Configuration parameters can be set for KSQL queries, the Kafka Admin Client, the KSQL Server, the Kafka Streams API, the Kafka Producer, and the Kafka Consumer. 
 
-You can set the following parameters for the KSQL Server.
+These can all be configured via the ``/etc/ksql/ksql-server.properties`` file. 
 
 .. important:: KSQL Server configuration settings take precedence over those set in the KSQL CLI. For example, if a value
                for ``ksql.streams.replication.factor`` is set in both the KSQL Server and KSQL CLI, the KSQL Server value is used.
 
-KSQL Query
+KSQL Query Parameters
```

Comment:
This is as odd term for something like `ksql.service.id` and `ksql.persistent.prefix`. These don't really control how a query executes.

What does the latter even do. 

I think we need a place to list _all_ ksql specific configuration parameters and what they do. That seems to be missing in the current docs.

Reply:
Agreed, though said page should be built from the code.

Reply:
Yea. This is a painful gap at the moment. I asked around, and it looks like we don't have a ready pipeline to take the documentation from the code and translate it to rst files. I think this will be worth the investment though. I filed #1076 to track this.

Reply:
@apurvam what are the options that it can currently translate to?

Reply:
Not really sure. The problem is that some of the settings are relevant to the query and some are not. I good general name doesn't come to mind :(

Reply:
@apurvam minimally can we point people to the internal KSQL source file that contains these configs?

Reply:
These are more KSQL engine parameters than parameters for individual queries. Some of the engine parameters control how individual queries are executed. ``ksql.sink.partitions`` for example controls how many partitions a sink topic has. ``ksql.service.id `` ``ksql.persistent.prefix`` are used to build the app name for the streams app.

Reply:
Feels like these need breaking down into categories, both in the docs and in the code, e.g.`KsqlAppConfig`, `KsqlQueryConfig`, etc.  These can all be merged into a single config, in a similar way to the how way the SSL & SASL config are brought into `ProducerConfig`.

Reply:
Suggestion: In the light of this, I'd recommend to:

- Merge two sections "KSQL Query Parameters" (which currently e.g. talks about `ksql.service.id`) and "KSQL Server" into one section "KSQL General Settings" for the time being. As long as we lack a more precise overview/categorization, we can then verbally mention that some of these settings such as `ksql.service.id` are used to configure the server/engine, whereas others like `ksql.sink.partitions` are blabla.
- Rename the "Admin Client" section into "KSQL Admin Client".

Reply:
Totally agree we should categorize the configs in KsqlConfig. This will also help us filter and whitelist settings that can be get/set via the REST API.

> Rename the "Admin Client" section into "KSQL Admin Client"
I think we should just get rid of the Admin Client section and call out that its one of the Kafka Clients that's configured according to the ksql.streams settings.